File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cs
public class Solution {
    public string ReverseVowels(string s) {
         string vowel = "aeiouAEIOU";
 int l = 0, r = s.Length - 1;
 char[] rst = s.ToCharArray();
 while(l < r)
 {
     if (vowel.Contains(s[l]) && vowel.Contains(s[r]))
     {

         char c = s[l];
         rst[l] = s[r];
         rst[r] = c;
         r--;
         l++;
     }else if (vowel.Contains(s[l]) && !vowel.Contains(s[r]))
     {
         r--;
     }
     else
     {
         l++;
     }
 }
 string rs = new string(rst);
 return rs;
    }
}

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {
    private void swap(char[] chars, int i, int j){
        char temp = chars[i];
        chars[i] = chars[j];
        chars[j] = temp;
    }
    public String reverseVowels(String s) {
        boolean isVowels[] = new boolean[128];
        char[] chars = s.toCharArray();
        for(char c : "aeiouAEIOU".toCharArray())
            isVowels[c] = true;

        int st = 0, en = s.length() - 1;

        while (st < en){
            if(!isVowels[chars[st]]){
                st++;
                continue;
            }
            if(!isVowels[chars[en]]){
                en--;
                continue;
            }
            swap(chars,st++,en--);
        }
        return new String(chars);
    }
}